src/ecalc/libraries/libecalc/common/libecalc/core/consumers/consumer_system.py

```diff
@@ -180,29 +191,44 @@ def evaluate(
                         regularity=temporal_operational_settings.models[0].model[0].rates[0].regularity,
                     ),
                 )
+
+                energy_usage[valid_indices] = energy_usage_for_period[valid_indices_for_period].values
                 power[valid_indices] = power_for_period[valid_indices_for_period].values
+                operational_settings_used[valid_indices] = [operational_setting_index + 1 for _ in valid_indices]
+
+                # Storing all consumer results for all calculated operational settings.
+                all_consumer_results[str(period)][operational_setting_index] = consumer_results
 
-                operational_settings_used[valid_indices] = operational_setting_index + 1  # 1-based index
+                # Fixme: Unit data should come from the consumer itself, not the results.
+                energy_usage.unit = consumer_results[0].component_result.energy_usage.unit
+                power.unit = consumer_results[0].component_result.power.unit
```

Comment:
can this fail if it is not set? or is it always set?

Reply:
This is handled in a separate PR.